torch/fx/experimental/proxy_tensor.py

```diff
@@ -108,13 +108,13 @@ def has_proxy(obj):
 
 def set_meta(proxy, val):
     if isinstance(val, FakeTensor):
-        proxy.node.meta['val'] = val
+        proxy.node.meta['val'] = val.detach()
```

Comment:
What if val is a constant

Reply:
@SherlockNoMad all of the requires_grad is False for all the val in the graph

Reply:
Cannot use val to drive interpreter anymore.

Reply:
So it turns out that it is pretty difficult to get accurate `requires_grad` metadata on the fake tensors if you detach. This is because, at the point proxy tensor is running (under autograd), we don't actually have accurate requires grad metadata yet! It gets set correctly on the tensor when we return from proxy tensor. If we save the fake tensor at this point in time, we don't get the correct setting reflected on the fake tensor when it is modified in place.

Reply:
So I'm going to skip requires_grad recording and file a bug for it for later, if anyone cares.